### DISCUSSION TITLE
Make Action Cache public accessible

### DISCUSSION THREAD

**Discussion by hfhbd** [community/community#44567] [Actions]
### Select Topic Area

Product Feedback

### Body

The Action Cache is a very nice feature to speed up build running on GitHub runners using build caching features from the build system, eg https://docs.gradle.org/current/userguide/build_cache.html

But the cache is currently only available for GitHub runners due to internal access. It would be very handy to expose the build cache as a (write-able or) read-only API too. 
This allows developers to reuse the cache on local environments reducing running the build due to cached build results.
This also affects GitHub Codespaces which does not reuse the action cache at the moment, reducing costs.

I think the current APIs don't require a change, you only need to make them public: https://github.com/nektos/act/issues/329

**Comment by bishal-pdMSFT** [comment]
@hfhbd Actions cache is primarily targeted for speeding up workflow runs. I would suggest using Packages feature if you have needs to use build artifacts outside of Actions realm.

**Reply by ruffsl** [reply]
Would you recommend using Packages features for primarily caching and speeding up GitHub Codespaces runs? I would suggest using build artifacts outside the Action realm would not be as appropriate for continuous development.
